mothernet/priors/step_function_prior.py

```diff
@@ -0,0 +1,46 @@
+import torch
+
+from mothernet.distributions import parse_distributions, sample_distributions
+from mothernet.utils import default_device
+
+
+class StepFunctionPrior:
```

Comment:
can you please add a unit test, maybe just that it respects the number of features and samples? testing it has a step per feature also shouldn't be too hard, right?